src/main/java/com/teragrep/pth_06/planner/HdfsDBClient.java

```diff
@@ -0,0 +1,121 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.planner;
+
+import com.teragrep.pth_06.FileSystemFactoryImpl;
+import com.teragrep.pth_06.config.Config;
+import com.teragrep.pth_06.HdfsTopicPartitionOffsetMetadata;
+import org.apache.hadoop.fs.*;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.LinkedList;
+
+// Searches the HDFS database for files contained in directories based on topicsRegexString. Creates a list of objects where each object contains the metadata of a single HDFS-file.
+public class HdfsDBClient {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(HdfsDBClient.class);
+    private final String path;
+    private final FileSystem fs;
+    private final String topicsRegexString;
+    private final long ignoreBeforeEpoch;
+
+    public HdfsDBClient(Config config, String topicsRegexString) throws IOException {
+        this.topicsRegexString = topicsRegexString;
+        this.ignoreBeforeEpoch = config.hdfsConfig.includeFileEpochAndAfter; // Defines the minimum time window / cutoff epoch for making sure that the files which metadata is fetched by the planner are not mistakenly deleted by cfe_39 pruning before tasker can process the records inside.
+        path = config.hdfsConfig.hdfsPath;
+        fs = new FileSystemFactoryImpl(config.hdfsConfig).create(true);
```

Comment:
create primary constructor that has FileSystem as a parameter, and make this a secondary constructor that calls the primary constructor with `this(..., new FileSystemFactoryImpl(config.hdfsConfig).create(true))`

Reply:
This allows providing custom FileSystem if necessary.

Reply:
Solved in commit a59ebd2b03ababb2317138b89208c0b001888c67